admin-console/src/main/webapp/src/components/CreateEditModal.tsx

```diff
@@ -0,0 +1,75 @@
+
+interface CreateEditModalProps {
+    isEdit: boolean;
+    showModal: boolean;
+    applicationName?: string;
+    eventTypeName?: string;
+    eventTypeDisplayName?: string;
+    eventTypeDescription?: string;
+    onClose: () => void;
+    onSubmit: () => void;
+    eventTypeQuery: unknown;
+    onChange: (value: string, event: React.FormEvent<HTMLInputElement> | React.FormEvent<HTMLTextAreaElement>) => void;
+
+}
+
+export const CreateEditModal: React.FunctionComponent<CreateEditModalProps> = (props) => {
+    const createNewEvent = useCreateEventType();
+
+    return (
+        <React.Fragment>
+            <Modal
+                variant={ ModalVariant.medium }
+                title={ `${ props.isEdit ? 'Update' : 'Create'} Event Type for ${ props.applicationName }` }
+                isOpen={ props.showModal }
+                onClose={ props.onClose }
+            ><Form isHorizontal>
+                    <FormGroup label='Name' fieldId='name' isRequired
+                        helperText={ props.isEdit ? <HelperText><HelperTextItem variant="warning" hasIcon>
+                                                    If this field is modified it may affect exisiting behavior.
+                        </HelperTextItem></HelperText> : 'This is a short name, only composed of a-z 0-9 and - characters.' }>
+                        <TextInput
+                            type='text'
+                            value={ props.eventTypeName }
+                            onChange={ props.onChange }
+                            id='name'
+                            name="name"
+                        /></FormGroup>
+                    <FormGroup label='Display name' fieldId='display-name' isRequired
+                        helperText='This is the name you want to display on the UI'>
+                        <TextInput
+                            type='text'
+                            value={ props.eventTypeDisplayName }
+                            onChange={ props.onChange }
+                            id='display-name'
+                            name="displayName"
+                        /></FormGroup>
+                    <FormGroup label='Description' fieldId='description'
+                        helperText='Optional short description that appears in the UI
+                                                to help admin descide how to notify users.'>
+                        <TextArea
+                            type='text'
+                            value={ props.eventTypeDescription }
+                            onChange={ props.onChange }
+                            id='description'
+                            name="description"
+                        /></FormGroup>
+                    <ActionGroup>
+                        <Button variant='primary' type='submit'
+                            { ...(createNewEvent.loading || createNewEvent.payload?.status !== 200) ?
```

Comment:
I don't think this is doing what we need. `createEvent` in this component is never used. We could pass a `loading` prop to this component.